### PR TITLE
Updating GASETH UMIP with an additional calculation implementation

### PR DESCRIPTION
**Pull request by smb2796**
Using a Google BigQuery query that @mrice32 created,  I have added another reference implementation. This will allow for flexibility in the actual usage of this price identifier. I also attempt to make it explicit that DVM reporters are free to define their own implementations of this calculation later, so long as those implementations agree with the calculation methodology.

### CODE REVIEW

File: UMIPs/umip-16.md [hunk 1]
```diff
@@ -91,6 +91,48 @@ def return_median(t, N)
...
+            FROM
+                `bigquery-public-data.crypto_ethereum.transactions`
+            WHERE
+                block_timestamp BETWEEN TIMESTAMP(t2)
+                AND TIMESTAMP(t1)
+            GROUP BY
+                gas_price,
+                block_number
+        )
+);
```
Comment (mrice32) [inline UMIPs/umip-16.md:125]: Love the formatting! What tool did you use?

File: UMIPs/umip-16.md [hunk 2]
```diff
@@ -91,6 +91,48 @@ def return_median(t, N)
...
+  
+set halfway = (SELECT DIV(MAX(cum_sum),2) from cum_gas);
+
+set block_amount = (SELECT (MAX(block_number) - MIN(block_number)) FROM cum_gas);
+
+select cum_sum, gas_price from cum_gas where cum_sum > halfway order by gas_price limit 1;
```
Comment (mrice32) [inline UMIPs/umip-16.md:131]:
```suggestion
SELECT cum_sum, gas_price FROM cum_gas WHERE cum_sum > halfway ORDER BY gas_price LIMIT 1;
```

**Review by mrice32 — APPROVED** [review]
LGTM!

File: UMIPs/umip-16.md (hunk 1 above)
Comment (smb2796) [reply UMIPs/umip-16.md:125]: Thanks! Found a cool VSCode extension called SQL Formatter -> https://marketplace.visualstudio.com/items?itemName=adpyke.vscode-sql-formatter

Comment (daywiss) [reply UMIPs/umip-16.md]: theres also one built into cloud ui "format"
image

Comment (smb2796) [reply UMIPs/umip-16.md]: Found that one afterwards. It's much better than SQL Formatter

**Review by mrice32 — APPROVED** [review]
LGTM! Btw, I'm assuming this is only marginally more expensive than the existing logic, right? If not, we may want to optimize a bit more.

**Review by chrismaree — APPROVED** [review]
This is awesome! love that we can do it in google big query. Also, the fact that you get 300 USD + 1 TB of free quota means that this is accessible for voters and bot runners alike. Good work @smb2796

**Pull request merged**